### DISCUSSION TITLE
How to delete just the draft version of a data set?

### DISCUSSION THREAD

**Discussion by mankoff** [gdcc/pyDataverse#144] [Q&A] [answered]
About 10 % of the time when I publish a dataset with `resp = api.publish_dataset(identifier, "major")`, nothing happens. It stays in draft mode. I assume this may be our DOI provider timing out or something. Anyway, when I update the data the next day, I need to test if it is somehow 'stuck' in draft mode, and if so, delete the draft before uploading new files. I check for `draft` mode with `if resp.json()['data']['latestVersion']['versionState'] == 'DRAFT':`, but now I'm not sure what to do. What is the API command to delete a draft dataset (but not the real published dataset)?

**Comment by pdurbin** [comment, marked as answer]
@mankoff you can't delete a published dataset unless you use the "destroy" API endpoint (and it's superuser-only). Please see https://guides.dataverse.org/en/5.10/api/native-api.html#delete-published-dataset
